rust/ip-packet/src/proptests.rs

```diff
@@ -0,0 +1,104 @@
+use std::net::{IpAddr, Ipv4Addr, Ipv6Addr};
+
+use pnet_packet::Packet;
+use proptest::arbitrary::any;
+use proptest::strategy::Strategy;
+
+use crate::make::{icmp_reply_packet, icmp_request_packet, tcp_packet, udp_packet};
+use crate::MutableIpPacket;
+
+fn non_icmp_packet() -> impl Strategy<Value = MutableIpPacket<'static>> {
+    (
+        any::<IpAddr>(),
+        any::<IpAddr>(),
+        any::<u16>(),
+        any::<u16>(),
+        any::<Vec<u8>>(),
+        any::<u8>(),
+    )
+        .prop_filter("IPs must be of the same version", |(src, dst, ..)| {
+            src.is_ipv4() == dst.is_ipv4()
+        })
+        .prop_map(|(src, dst, sport, dport, payload, proto)|
+            // using a bool instead of prop_oneof because we can't wrap mutableippacket in Just because it's non-clonable
```

Comment:
But you could make dedicated strategies like this:

```rust
fn udp_packet() -> impl Strategy {
	(any::<IpAddr>(), ..).prop_map(|(src)| ip_packet::make::udp_packet())
}
```

I think that would be more modular.